graphs/check_bipartite_graph_bfs.py

```diff
@@ -10,6 +10,24 @@
 
 
 def check_bipartite(graph):
+    """
+    >>> check_bipartite(
+    ... {0: [1, 3], 1: [0, 2], 2: [1, 3], 3: [0, 2]}
+    ... )
+    True
+    >>> check_bipartite(
+    ... {0: [1, 2, 3], 1: [0, 2], 2: [0, 1, 3], 3: [0, 2]}
+    ... )
+    False
+    >>> check_bipartite(
+    ... {0: [4], 1: [], 2: [4], 3: [4], 4: [0, 2, 3]}
+    ... )
+    True
+    >>> check_bipartite(
+    ... {0: [1, 3], 1: [0, 2], 2: [1, 3], 3: [0, 2], 4: [0]}
+    ... )
+    False
```

Comment:
Please add some negative values, floating point values, and some strings.
Also,
`{7: [1, 3], 1: [0, 2], 2: [1, 3], 3: [0, 2], 4: [0]}`
`{0: [1, 3], 1: [0, 2], 2: [1, 3], 3: [0, 2], 9: [0]}`

Reply:
thanks for the additional test cases! I'll have to refactor the code a bit as it assumes that there's keys for all values from 0 to n. Will add different data types as well.

Reply:
It is OK to fail if the input is not 1 to n or if other garbage data is provided.  We just want to ensure the code does safe things in the face of garbage data.  Raising a ValueError is safe.

Reply:
Understood. Thanks for the clarification!